### DISCUSSION TITLE
Moving to `ue5-style-guide`

### DISCUSSION THREAD

**Discussion by Allar** [Allar/ue4-style-guide#65] [Announcements]
This style guide is no longer modern enough for Gamemakin LLC (mainly just me, Michael Allar) for use with Unreal Engine 5.

All work on the upgrade to v2 / UE5 will be done on the v2 branch until it is ready to be moved to main.

I think the best thing moving forward for everyone who believes in and uses this document in one form or the other is to make a push to get this updated using the knowledge we've all learned over the years and things we're still discovering about Unreal Engine 5.

Looking for feedback of all kinds, including but not limited to:

- Nanite
- Lumen
- Virtual Production
- Metasounds
- Metahumans
- Quixel Megascans
- Raytracing
- World / Level Building
- Gameplay Ability System
- C++ Code Style
- Literally Everything

I would like to make conversation about how we should all be doing things happen. Feel free to join the Discord linked in the main readme, create Github Discussions, tweet, whatever you can do.

Don't know how to start helping? Take a look at the Issues page for some ideas. https://github.com/Allar/ue5-style-guide/issues

**Comment by miltoncandelero** [comment]
I do have an idea, can we use `MM_` for Master Materials? It would be an easy way to mark that the material is meant to be Instantiated and not used as-is 🤔

**Reply by Allar** [reply]
This is a great idea. Moving to issue https://github.com/Allar/ue5-style-guide/issues/68

**Reply by CasulScrub** [reply]
How about MB_ for material base instead of Master.

**Comment by RaeASunshine** [comment]
Some best practices around Modular Gameplay would probably be good to have.  Building content to be modular is probably a "full project" kind of thing, and many people use this style guide as a bible for how to organize a project.

**Reply by Allar** [reply]
I've dove very deep into some modular gameplay concepts, would love to see how other people are doing it though.

**Reply by RaeASunshine** [reply]
Same.  I've been starting to port some of my plugins to more modular concepts, utilizing the Data Registry and stuff.  I'm very interested in best practice discussion around that and coming to a consensus on the best way to do it

**Reply by doublebuffered** [reply]
The engine code supporting modular gameplay is changing pretty rapidly right now so I would recommend letting it settle for 5.0 before finalizing anything about project structure

**Comment by Tylrin** [comment]
There are also new input classes in the enhanced input plugin, like input action `IA_`  and input mapping context `IM_` or `IMC_`. The prefixes are suggestions.

**Reply by Allar** [reply]
Thanks for bring this up!

**Comment by SVasilev** [comment]
Looking at the **Lyra** example project I see that blueprints are now starting with `B_`
I think this example project answers a lot of questions in terms of abbreviations and other things, as well. I would suggest it is a good starting point when updating the style guide to UE5 :)

In fact, I might as well add this as a proper issue in the v2 branch.

**Reply by KavanBahrami** [reply]
Lyra also uses W_ for Widget Blueprints, rather than the suggested WBP_ from this style guide.

I've started using both B_ and W_ and will likely continue. The fewer characters needed to express the context the better in my book, esp with longer descriptive names being my preference.

**Comment by Jamie-Gordon300** [comment]
Hello! I have recently started looking into Metasounds for a project i am working on. Currently when scanning it suggests the prefix `A_ ` which is the same as audio assets. I think it might be useful to have them as `MSS_` for Metasound Sources and `MSP_` for Metasound Patches.
